### PR TITLE
Add bumper pubs, remove obsolete debug pubs.

### PR DESCRIPTION
**Pull request by markwomack**

### CODE REVIEW

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino [hunk 1]
```diff
@@ -439,6 +448,38 @@ void doPublishRawImu() {
...
+
+  double bumperLeft = MAX_FORCE_LEFT_BUMPER 
+                        * ((rawBumperLeft - MIN_RAW_LEFT_BUMPER)
+                            /((double)(MAX_RAW_LEFT_BUMPER - MIN_RAW_LEFT_BUMPER)));
+  double bumperRight = MAX_FORCE_RIGHT_BUMPER 
+                        * ((rawBumperRight - MIN_RAW_RIGHT_BUMPER)
+                            /((double)(MAX_RAW_RIGHT_BUMPER - MIN_RAW_RIGHT_BUMPER)));
+
+  ros_bumper_left_msg.header.stamp = current_time;
+  // TODO(mwomack): set header frame to what?
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino]: Set the frames to /bumper_left and /bumper_right.
We'll have to define the transformation in the urdf.

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino [hunk 2]
```diff
@@ -439,6 +448,38 @@ void doPublishRawImu() {
...
+ */
+void doPublishBumpers(void) {
+  ros::Time current_time = ros_nh.now();
+
+  int rawBumperLeft = analogRead(BUMPER_LEFT_PIN);
+  int rawBumperRight = analogRead(BUMPER_RIGHT_PIN);
+
+  double bumperLeft = MAX_FORCE_LEFT_BUMPER 
+                        * ((rawBumperLeft - MIN_RAW_LEFT_BUMPER)
+                            /((double)(MAX_RAW_LEFT_BUMPER - MIN_RAW_LEFT_BUMPER)));
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino]: It's preferable to use `static_cast<double>(MAX ...)`

To shorten the line width, you could 
```
static const int RANGE_RAW_LEFT_BUMPER(MAX_RAW_LEFT_BUMPER - MIN_RAW_LEFT_BUMPER);
```

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino [hunk 3]
```diff
@@ -546,14 +568,14 @@ void doTeensyDebug() {
...
- * or 100 hz = 10 millis.
+/**
+ * Sets up a callback that is called at the
+ * given frequency.
  */
-int convertFreqToMillis(int frequency) {
-  return static_cast<int>((1.0/frequency) * 1000);
+void setupTimerCallback(int callback_freq_hz, void (*callback)(void)) {
+  // Convert frequency in hertz to milliseconds
+  int callback_millis = static_cast<int>((1.0/callback_freq_hz) * 1000);
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino]: If you wanted to save one cycle you could do
```
static_cast<int>(1000.0/callback_freq_hz);
```

**Review by griswaldbrooks — CHANGES_REQUESTED** [review]

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino [hunk 4]
```diff
@@ -546,14 +568,14 @@ void doTeensyDebug() {
...
- * number of milliseconds to match the
- * frequency. For example, 2 hz = 500 millis
- * or 100 hz = 10 millis.
+/**
+ * Sets up a callback that is called at the
+ * given frequency.
  */
-int convertFreqToMillis(int frequency) {
-  return static_cast<int>((1.0/frequency) * 1000);
+void setupTimerCallback(int callback_freq_hz, void (*callback)(void)) {
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino:577]: `setupTimerCallback` needs to be forward declared for `setup` to recognize it, according to travis.

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino (hunk 2 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino]: Done. Made the range a double, so I don't need to cast.

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino (hunk 1 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino]: Done.

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino (hunk 3 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino]: Done.  I don't think any cast is required.

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino (hunk 4 above)
Comment (markwomack) [reply Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino:577]: Done

File: Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino [hunk 5]
```diff
@@ -574,7 +576,7 @@ void doTeensyDebug() {
  */
 void setupTimerCallback(int callback_freq_hz, void (*callback)(void)) {
   // Convert frequency in hertz to milliseconds
-  int callback_millis = static_cast<int>((1.0/callback_freq_hz) * 1000);
+  int callback_millis = (1000/callback_freq_hz);
```
Comment (griswaldbrooks) [inline Arduino/victoria_teensy_ros_node/victoria_teensy_ros_node.ino]: 1000.f

**Pull request merged**